### PR TITLE
[BugFix] BE crashes on illegal JSON

### LINKED ISSUE #12816: BE crash on illegal json
### Steps to reproduce the behavior (Required)
1. Create table
```
create table tbl_json (
	id int ,
	name string ,
	score int
)
duplicate key(id) 
distributed by hash(id) buckets 1
PROPERTIES (
  "replication_num" = "1"
)
```
1. Do stream load
```
a.json
{
{"id":"5","name":"Lucy","score":22},
{"id":"6","name":"Hyk","score":27}
}
```
```
curl --location-trusted -u root:  'http://127.0.0.1:18082/api/db0/tbl_json/_stream_load' \-X PUT \-H 'strip_outer_array: false' \-H 'format: json' -T a.json
```

### Expected behavior (Required)
Load is OK.
### Real behavior (Required)
BE crash.
```
*** Aborted at 1667192760 (unix time) try "date -d @1667192760" if you are using GNU date ***
PC: @          0x27460a1 starrocks::vectorized::JsonDocumentStreamParser::get_current()
*** SIGSEGV (@0x8) received by PID 12653 (TID 0x7fa1a94c1700) from PID 8; stack trace: ***
    @          0x3fa3ad2 google::(anonymous namespace)::FailureSignalHandler()
    @     0x7fa2a1087630 (unknown)
    @          0x27460a1 starrocks::vectorized::JsonDocumentStreamParser::get_current()
    @          0x27455d7 starrocks::vectorized::JsonReader::_read_rows<>()
    @          0x27414d9 starrocks::vectorized::JsonReader::read_chunk()
    @          0x27416ec starrocks::vectorized::JsonScanner::get_next()
    @          0x272e5e0 starrocks::vectorized::FileScanNode::_scanner_scan()
    @          0x272ff4f starrocks::vectorized::FileScanNode::_scanner_worker()
    @          0x5a21410 execute_native_thread_routine
    @     0x7fa2a107fea5 start_thread
    @     0x7fa2a069ab0d __clone
```
### StarRocks version (Required)
2.4.0, 2.3.3, 2.2.8

### PR DESCRIPTION
**Pull request by rickif**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/starrocks/issues/12816

## Problem Summary(Required) ：

When the input is invalid JSON, like `{"key": 1} "key": 2}`, the 2nd document would not be parsed successfully. In order to get the context that leads the parsing failure, we use `JsonFunctions::to_json_string(doc)` to print the content of the doc. But for doc that is not parsed successfully, the  `JsonFunctions::to_json_string(doc)` would lead the BE crash.
This PR removes the printing of not parsed doc. The BE would record the current parsed location, and print the not-parsed content to the error log. In this way, not only the bug leading the crash would be resolved, but also the user is easier to get to know the error in the json input.
Besides, this PR would not change the behavior of the SR. It would modify the error message and adds more content to the error log file.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by rickif** [comment]
The output would be like this:
```
curl "http://172.26.92.226:18082/api/_load_error_log?file=error_log_b042825f98f8bb3f_7c38c767043b5bb3"
Error: Data quality error: construct row in object order failed, error: The JSON document has an improper structure: missing or superfluous commas, braces, missing keys, etc.. Row: {
{
Error: Data quality error: Failed to iterate document stream as object. error: The JSON element does not have the requested type.. Row: not parsed: "id":"5","name":"Lucy","score":22},
{"id":"6","name":"Hyk","score":27}
}
```

**Comment by rickif** [comment]
run starrocks_clang-tidy

File: be/src/exec/vectorized/json_parser.cpp [hunk 1]
```diff
@@ -72,6 +74,10 @@ Status JsonDocumentStreamParser::advance() noexcept {
     return Status::EndOfFile("all documents of the stream are iterated");
 }
 
+std::string JsonDocumentStreamParser::left_bytes_string(size_t sz) noexcept {
+    return std::string(reinterpret_cast<char*>(_data) + _off, std::min(_len - _off, sz));
```
Comment (xiaoyong-z) [inline be/src/exec/vectorized/json_parser.cpp:78]: how about use _doc_stream_itr.current_index() here as _off like json array parser

**Review by meegoo — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by rickif** [comment]
run starrocks_admit_test

**Comment by rickif** [comment]
run starrocks_admit_test

**Comment by rickif** [comment]
run starrocks_admit_test

**Comment by rickif** [comment]
run starrocks_admit_test

**Comment by rickif** [comment]
run all

**Pull request merged**